src/libbson/src/bson/bson-compat.h

```diff
@@ -332,6 +332,13 @@ typedef signed char bool;
 #define BSON_IF_MSVC(...)
 /** Expands the arguments if compiling with GCC or Clang, otherwise empty */
 #define BSON_IF_GNU_LIKE(...) __VA_ARGS__
+#elif defined(_AIX)
+/** Expands the arguments if compiling with MSVC, otherwise empty */
+#define BSON_IF_MSVC(...)
+/** Expands the arguments if compiling with GCC or Clang, otherwise empty */
+#define BSON_IF_GNU_LIKE(...) 
+/** Expands the arguments if compiling with AIX, otherwise empty */
+#define BSON_IF_GNU_LEGACY_ATOMICS(...) __VA_ARGS__
```

Comment:
The `BSON_IF_GNU_LEGACY_ATOMICS` is only used within `bson-atomic.h`, and is already defined if `BSON_USE_LEGACY_GCC_ATOMICS` is defined, so it need not be defined here.

If I understand correctly: If you are using a recent version of xlC, the branch for `defined(__clang__)` should be taken on line 330. If you are using a legacy version, a `|| defined(__xlC__)` may need to be added on line 330. This should add the necessary `#define BSON_IF_MSVC` that will suppress its appearance in `bson-atomic.h`.